### PR TITLE
Implemented ELK layout engine

### PR DESCRIPTION
**Pull request by spoenemann**

### CODE REVIEW

File: sprotty-elk/tsconfig.json [hunk 1]
```diff
@@ -1,12 +1,29 @@
...
+        "declaration": true,
+        "declarationMap": true,
+        "noImplicitAny": true,
+        "noImplicitThis": true,
+        "noImplicitReturns": true,
+        "noUnusedLocals": true,
+        "strictNullChecks": true,
+        "experimentalDecorators": true,
+        "emitDecoratorMetadata": true,
+        "jsx": "react"
```
Comment (JanKoehnlein) [inline sprotty-elk/tsconfig.json]: De we need tsx?

File: sprotty-elk/package.json [hunk 2]
```diff
@@ -27,22 +27,26 @@
...
   "devDependencies": {
+    "@types/chai": "^4.1.7",
+    "@types/mocha": "^5.2.5",
     "chai": "^4.2.0",
     "jenkins-mocha": "^6.0.0",
     "mocha": "^5.2.0",
+    "reflect-metadata": "^0.1.12",
     "rimraf": "^2.6.2",
-    "ts-node": "^7.0.1",
+    "ts-node": "<7.0.0",
```
Comment (JanKoehnlein) [inline sprotty-elk/package.json]: Is that constraint still necessary?

File: sprotty-elk/tsconfig.json (hunk 1 above)
Comment (spoenemann) [reply sprotty-elk/tsconfig.json]: Currently not, but it shouldn't hurt to keep the setting here.

File: sprotty-elk/package.json (hunk 2 above)
Comment (spoenemann) [reply sprotty-elk/package.json]: Yes, mocha tests did not work with ts-node 7.0.1.

**Pull request merged**